README
/*
 * $RCSfile: README,v $ $Revision: 1.5 $
 * $Author: slim_lv $ $Date: 2011/01/06 08:28:50 $
 * This file is part of CYRUP project
 * by Yuri Pimenov (up@msh.lv) & Deniss Gaplevsky (slim@msh.lv)
 */

Introduction
------------
CyrUp is web interface for mail system based on Postfix, Cyrus and MySQL/PostgreSQL software

Prerequisites
-------------
To use this software, you need the following stuff:
 - A *NIX (or similar) system, with PAM facility (when non-plain text passwords are used) is set up and working
 - A MySQL or PostgreSQL server, up and running
 - Postfix MTA 
 - Cyrus imap
 - Cyrus SASL (when non-plain text passwords are used)
 - Http server with PHP4/5 support 
 - Net_Sieve PEAR class for autoreply support

Installation instruction
------------------------
See INSTALL file for detail.

Notes (known issues)
----------------------
1. Domain suffix (when MAILBOX_STYLE is "USERSUFFIX") can be set at new domain creation time only (in Domains tab) - there is no posibility to change domain suffix after domain creation.
2. When alias is renamed (in Aliases tab) and this alias is in any maillist (in Maillists tab) - alias in maillist will not be renamed automaticly.
3. When alias is aliased to email (in Aliases tab), destinations email need to be separeted with ','.
4. When adding new account (in Accounts tab) and 'Autocreate alias?' is checked, CYRUP checks existence of corresponding alias, but silently drop alias creation if alias already exist.


CHANGELOG
/*
 * $RCSfile: CHANGELOG,v $ $Revision: 1.17 $
 * $Author: slim_lv $ $Date: 2016/11/01 14:16:45 $
 * This file is part of CYRUP project
 * by Yuri Pimenov (up@msh.lv) & Deniss Gaplevsky (slim@msh.lv)
 */

20161101 <slim_lv@users.sourceforge.net>
  bugfix: always use trailing slash with BASE_URL

20120903 <slim_lv@users.sourceforge.net>
    new: allow to disable whole domain to receive incoming mails. Users authorization via POP3/IMAP/SMTP is not affected

20110105 <slim_lv@users.sourceforge.net>
    new: autoreply can be configured for any account (require Net_Sieve PEAR class) - Roundcube/Avelsieve tested so far
    new: config option SHOW_VACATION_LIST allows to show autoreply status in list of accounts

20091025 <slim_lv@users.sourceforge.net>
    new: config option DEFAULT_DOMAIN allows usage of 'defaultdomain' in cyrus.conf with cyrup

20081222 <slim_lv@users.sourceforge.net>
    bugfix: fix php notice on undefined variables

20080828 <slim_lv@users.sourceforge.net> 
    new: new toolbar to filter listing in Accounts/Aliases/Maillists pages

20080826 <slim_lv@users.sourceforge.net> 
    new: config.inc.php - set SYSTEM_ALIASES to create aliases for common system users

20080430 <slim_lv@users.sourceforge.net> 
    bugfix: includes/integrity.inc.php - incorrect identification of CYRUS_USER user

20070622 <slim_lv@users.sourceforge.net>
    
    bugfix: includes/imap.inc.php - simplified mailbox renames code when using USER@DOMAIN.TLD scheme
    bugfix: includes/integrity.inc.php - checks are improved for USER@DOMAIN.TLD scheme
    new: INSTALL - new section added - how-to run without pam/saslauthd and with cleartext passwords 

20070513 <slim_lv@users.sourceforge.net>
    
    new: multi admin support is added

20070105 <slim_lv@users.sourceforge.net>
    
    new: config file (config.inc.php) is moved to root 
    bugfix: includes/imap.inc.php - incorrect imap string for creation IMAP folder when 
	    MAILBOX_STYLE = "USER@DOMAIN.TLD" 

20060818 <slim_lv@users.sourceforge.net>

    new: README - new section NOTES
    new: PostgreSQL support added, code by Brett Van Sprewenburg <brett(at)ataxxia.com>

20060204 <slim_lv@users.sourceforge.net>

    new: INSTALL - saslauthd options
    bugfix: accountform.php - unclosed HTML tag

20060202 <slim_lv@users.sourceforge.net>

    new: autocreate custom IMAP folders on account creation

20060201 <slim_lv@users.sourceforge.net>

    changes: config.php - SHA1() for passwords by default
    new: INSTALL - bayes options
    typo: INSTALL - in postfix config option

20060104 <slim_lv@users.sourceforge.net>

    bugfix: scripts/cyrup.mysql.sql -  DROP before CREATE

20051106 <slim_lv@users.sourceforge.net>

    typo: includes/config.inc.php - version number is 2.0 right now

20051103 <slim_lv@users.sourceforge.net>

    bugfix: scripts/cyrup.mysql.sql - incorrect fields number inserting admin's password

20051020 <slim_lv@users.sourceforge.net>
    
    2.0 pre
    install.php added
    bugfix: includes/mysql.inc.php

20051017 <slim_lv@users.sourceforge.net>
    
    entry version


config.default.inc.php
<?php
/*
 * $RCSfile: config.inc.php,v $ $Revision: 1.12 $
 * $Author: slim_lv $ $Date: 2016/11/01 14:19:46 $
 * This file is part of CYRUP project
 * by Yuri Pimenov (up@fs.lv) & Deniss Gaplevsky (slim@inbox.lv)
 */

# Primary configuration options. This is where you should start.

    define( 'ADMIN_USER', 'admin' );

    define( 'BASE_URL', 'http'.( isset($_SERVER['HTTPS']) ? 's' : '' ).'://'
                                                .$_SERVER['HTTP_HOST']
                                                .dirname($_SERVER['SCRIPT_NAME']) );     
					// Where the root of this project is visible from web

    define( 'INCLUDE_DIR', 'includes' );
    define( 'AUTHSIGN', "<a href='http://cyrup.sf.net' target='_blank'>cyrup.sf.net</a>" );


    define( 'DB_TYPE', 'mysqli' );   // Valid types are:
                                    //	"mysql" for MySQL (default)
                                    //  "mysqli" for MySQLi  
                                    //	"pgsql" for PostgreSQL
    # SQL
    define( 'SQL_HOST', 'localhost' );
    define( 'SQL_DB',   'mail' );
    define( 'SQL_USER', 'postfix' );
    define( 'SQL_PASS', 'mA1L' );

    # Cyrus
    define( 'CYRUS_HOST', 'localhost' );
    define( 'CYRUS_PORT', 143 );
    define( 'CYRUS_USER', 'cyrus' );
    define( 'CYRUS_PASS', 'itest' );

    define( 'CYRUS_DELIMITER', '/' );   // Default delimiter is '.'
                                        // if unixhierarchysep is set to 1 in imapd.conf
                                        // then delimiter must be '/'

    define( 'MAILBOX_STYLE', 'USER@DOMAIN.TLD' ); // Available: USERSUFFIX, USER@DOMAIN.TLD
                                                  // "USERSUFFIX"
                                                  //    Mailboxes will be created with the suffix 
                                                  //    specified for each domain.
                                                  //    For examaple "user_bogus", where suffix 
                                                  //    is "_bogus".
                                                  //    Suffix may be empty only for one domain.
                                                  // "USER@DOMAIN.TLD" (RECOMMENDED)
                                                  //    This style mailboxes works as 
                                                  //    virtual_mailbox_maps

    define( 'DEFAULT_DOMAIN', '' );       // when MAILBOX_STYLE is set to USER@DOMAIN.TLD it is possible to
                                          // set DEFAULT_DOMAIN same as defaultdomain in imapd.conf.
                                          // This works with saslauthd only

    define( 'PASSWORD_CRYPT', 4 );	// The same meaning as for pam_mysql crypt parameter
                                    // 0 - means clear text passwords,
                                    // 1 - UNIX crypt()
                                    // 2 - use MySQL's PASSWORD() function 
                                    //     NOT VALID FOR PGSQL
                                    //     (NOT RECOMMENDED)
                                    // 3 - MD5
                                    // 4 - SHA1 (RECOMMENDED)
                                    // See pam_mysql docs for details

# User interface & miscelaneous options

    define( 'SHOW_PASSWORD', false );   // true - one html "<input type=text ..." field 
                                        //  (usefull only with PASSWORD_CRYPT=0)
                                        // false - two html "<input type=password ..." 
                                        //  input fields (enter & reenter)

    define( 'MIN_PASSWORD_LENGTH', 4 ); // Less is not allowed
    define( 'DEFAULT_QUOTA', 50 );      // Default user quota in megabytes
    define( 'ALLOW_NO_QUOTA', 1 );      // While "true" admin can disable user's quota 
                                        //	by setting it to "0"
    define( 'SHOW_VACATION_LIST', 1 );  // Report vacation status in list of accounts (may be slow)

    define( 'DOMAIN_EXPORT_FILE', '/etc/postfix/local.domains' ); // Path to file where to save 
                                        // the list of domains (one per line).
                                        // May be used to speed-up domain search.
                                        // A webserver's user needs to have write permissions to this file.
                                        // Leave empty when not used.

    define( 'SYSTEM_ALIASES', '/etc/postfix/aliases.pcre' ); 
                                        // Export system domain names for system aliases ( postmaster etc )
                                        // Create file with correct permissions or leave blank to skip

    define ( 'SESS_LIFE' , get_cfg_var('session.gc_maxlifetime'));

    define ('IMAP_FOLDERS', 'SPAM, Sent ' );  // List of IMAP folders, separated by comma to be created for new accounts (but not subscribed)

    define( 'JS_URL', BASE_URL.'/js' );         // Where all JavaScript includes lie
    define( 'IMAGES_URL', BASE_URL.'/img' );    // Images

    define( 'VERSION', ' CyrUp v2.4.1' );         // Version. The bigger the better
    define( 'TITLE', VERSION.': ' );            // Prepend this to html title

    # Debug level & verbose messages
    define( 'D_NONE', 0 );           // no debug messages
    define( 'D_INCLUDE', 1 );        // include() or require()
    define( 'D_FUNCTION', 2 );       // most function calls
    define( 'D_SQL_ERROR', 4 );      // sql errors
    define( 'D_IMAP_ERROR', 8 );     // imap errors
    define( 'D_ALL', 255 );          // full verbose

    define ( 'DEBUG_LEVEL', D_SQL_ERROR | D_IMAP_ERROR );
#    define ( 'DEBUG_LEVEL', D_ALL );
#    define ( 'DEBUG_LEVEL', D_SQL_ERROR | D_FUNCTION | D_INCLUDE );

?>
